nms/app/packages/magmalte/app/views/subscriber/SubscriberChart.js

```diff
@@ -0,0 +1,330 @@
+  subscriberId: subscriber_id,
+  start: moment,
+  end: moment,
+  enqueueSnackbar: (msg: string, cfg: {}) => ?(string | number),
+};
+
+const getPromValue = resp => {
+  const respArr = resp?.data?.result
+    ?.map(item => {
+      return parseFloat(item?.value?.[1]);
+    })
+    .filter(Boolean);
+  return respArr && respArr.length ? respArr[0] : 0;
+};
+
+function getLabelUnit(val: number) {
+  if (val > 1000000) {
+    return [(val / 1000000).toFixed(2), 'mb'];
+  } else if (val > 1000) {
+    return [(val / 1000).toFixed(2), 'kb'];
+  }
+  return [val, 'bytes'];
+}
+
+async function getDatasets(props: DatasetFetchProps) {
+  const {start, end, networkId, subscriberId} = props;
+  const [delta, unit, format] = getStep(start, end);
+  let requestError = '';
+  const step = getStepString(delta, unit);
+  const toolTipHint = delta + ' ' + unit;
+
+  const queries = [
+    {
+      q: `sum(increase(ue_traffic{IMSI="${subscriberId}",direction="down"}[${step}]))`,
+      color: colors.secondary.dodgerBlue,
+      label: 'download',
+    },
+    {
+      q: `sum(increase(ue_traffic{IMSI="${subscriberId}",direction="up"}[${step}]))`,
+      color: colors.data.flamePea,
+      label: 'upload',
+    },
+  ];
+  const allDatasets = [];
+  const labels = [];
+  let addLabels = true;
+  const requests = queries.map(async (query, index) => {
+    try {
+      const resp = await MagmaV1API.getNetworksByNetworkIdPrometheusQueryRange({
+        networkId: networkId,
+        start: start.toISOString(),
+        end: end.toISOString(),
+        step: getStepString(delta, unit),
+        query: query.q,
+      });
+
+      const data = [];
+      resp.data.result.forEach(it =>
+        it['values']?.map(i => {
+          if (addLabels) {
+            labels.push(moment.unix(parseInt(i[0])).format(format));
+          }
+          data.push({
+            t: parseInt(i[0]) * 1000,
+            y: parseFloat(i[1]),
+          });
+        }),
+      );
+      addLabels = false;
+      allDatasets.push({
+        datasetKeyProvider: index.toString(),
+        label: query.label,
+        fill: false,
+        barPercentage: 0.7,
+        pointHitRadius: 10,
+        pointRadius: 2,
+        borderWidth: 2,
+        backgroundColor: query.color,
+        borderColor: query.color,
+        hoverBackgroundColor: query.color,
+        hoverBorderColor: 'black',
+        data: data,
+        unit: 'bytes',
+      });
+    } catch (error) {
+      requestError = error;
+      return [];
+    }
+  });
+
+  await Promise.all(requests);
+  if (requestError) {
+    props.enqueueSnackbar('Error getting event counts', {
+      variant: 'error',
+    });
+  }
+  return {allDatasets, labels, unit, toolTipHint};
+}
+
+function SubscriberDataKPI() {
+  const {match} = useRouter();
+  const networkId: string = nullthrows(match.params.networkId);
+  const subscriberId: string = nullthrows(match.params.subscriberId);
+  const [usage1H, setUsage1H] = useState([0, 'bytes']);
```

Comment:
Up to you: do you think it's cleaner to instead just have a single object with these fields instead of 4 separate state variables?

Reply:
I'm also not a huge fan of using a mixed array to store [value, unit]. Maybe at least make this an object